### ISSUE TITLE
Model.metrics_names not returning all metrics

### ISSUE THREAD

**Issue by pinsondg** [keras-team/keras#20244]
Backend: TF 2.16.1

I am building a model using the functional API and am compiling the model like so:
```python
model.compile(optimizer='adam', loss='binary_crossentropy', metrics=[
    keras.metrics.Precision(),
    keras.metrics.Recall(),
    keras.metrics.BinaryAccuracy(),
    ...
])
```
When I call `model.metrics_names` I expect to see all of my metrics returned (i.e. `['loss', 'precision', 'recall', 'binary_accuracy', ...]`) however, I only see `['loss', 'compile_metrics']`. It seems like all of my additional metrics are being grouped under `compile_metrics`, and it is making is difficult to actually know which metric is which when calling `model.evaluate()`.  This was never an issue before TF 2.16.1. Also, all of the metrics appear when training my model. 

Can someone help me understand how to get the complete list of metrics names?

**Comment by fchollet** [comment]
>  and it is making is difficult to actually know which metric is which when calling model.evaluate()

I recommend you just use `return_dict=True` in `evaluate()`, it will return a dict mapping metric name to value.

> Can someone help me understand how to get the complete list of metrics names?

You can also list the metrics grouped under compile metrics: list `model.metrics` and when you find the `CompileMetrics` object, list `m.metrics`

**Comment by pinsondg** [comment]
Oh nice, I missed that there was a return_dict argument. That solves my issue perfectly. Thanks @fchollet
